app/src/main/java/bankingapp/LogIn.java

```diff
@@ -75,8 +21,11 @@ public String getPinNum() {
      * @param pinNumber
      * @throws SQLException
      */
-    public void checkAccount(Connection con, String accountNumber, String pinNumber) throws SQLException {
-        if (DBTrans.checkAcc(con, accountNumber, Integer.parseInt(pinNumber))) {
+    public void checkAccount(DBTransactions db, Connection con, String accountNumber, String pinNumber)
```

Comment:
Notice, we're now in a position where we're pulling in a DBTransactions object exactly the same way we were previously doing with Connection objects in the DBTransactions class.

We _could_ change this in the same way, by pulling the DBTransactions instance into the constructor, and handling it once and for all at the class level.

However, I think it's more valuable to transition checkAccount to a static method of a static helper class, and leave it as is. That's subjective, and at the end of the day, it's your call.

Reply:
At the very least, pull in an `IDBTransactions` object, not a `DBTransactions` object. That allows writing a second, different implementation of IDBTransactions and allowing it to work seamlessly here.